src/UseCase/SendChangeUserEmailConfirmation.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * Copyright (C) 2018 Adshares sp. z o.o.
+ *
+ * This file is part of ADS Operator
+ *
+ * ADS Operator is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ADS Operator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ADS Operator.  If not, see <https://www.gnu.org/licenses/>
+ */
+
+namespace Adshares\AdsOperator\UseCase;
+
+use Adshares\AdsOperator\Mailer\Exception\CannotSendEmailException;
+use Adshares\AdsOperator\Mailer\MailerInterface;
+use Adshares\AdsOperator\Repository\Exception\UserNotFoundException;
+use Adshares\AdsOperator\Repository\UserRepositoryInterface;
+use Adshares\AdsOperator\Template\Exception\CannotFindTemplateException;
+use Adshares\AdsOperator\Template\TemplateInterface;
+use Psr\Log\LoggerInterface;
+
+class SendChangeUserEmailConfirmation
+{
+    /**
+     * @var UserRepositoryInterface
+     */
+    private $userRepository;
+
+    /**
+     * @var MailerInterface
+     */
+    private $emailProvider;
+
+    /**
+     * @var TemplateInterface
+     */
+    private $template;
+
+    /**
+     * @var string
+     */
+    private $templatePath;
+
+    /**
+     * @var string
+     */
+    private $senderEmail;
+
+    /**
+     * @var string
+     */
+    private $baseUrl;
+
+    /**
+     * @var string
+     */
+    private $subject;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function has 8 parameters, which is greater than the 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS107)